framework/src/main/java/org/checkerframework/common/basetype/BaseTypeValidator.java

```diff
@@ -712,4 +714,54 @@ public boolean areBoundsValid(
 
         return true;
     }
+
+    /**
+     * Validate if the annotations on wildcard are permitted by @TargetLocations. Report an error if
+     * the actual use of this annotation is not listed in the declared TypeUseLocations
+     * in @TargetLocation.
+     *
+     * @param type the type to check
+     * @param tree the tree of this type
+     */
+    protected void validateWildCardTargetLocation(AnnotatedWildcardType type, Tree tree) {
+        if (visitor.ignoreTargetLocation) return;
+
+        for (AnnotationMirror am : type.getSuperBound().getAnnotations()) {
+            List<TypeUseLocation> locations =
+                    visitor.AnnoToTargetLocations.get(AnnotationUtils.annotationName(am));
+            // @Target({ElementType.TYPE_USE})} together with no @TargetLocations(...) means
+            // that the qualifier can be written on any type use
```

Comment:
```suggestion
            // that the qualifier can be written on any type use.
```

Make comments full sentences and end with a period.